mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1168,141 +1261,49 @@ def _log_artifacts(self):
-        self._calculate_general_text_metrics()
-
-        self._log_eval_table()
-        name = _EVAL_TABLE_FILE_NAME.split(".", 1)[0]
-        self.artifacts[name] = JsonEvaluationArtifact(
-            uri=mlflow.get_artifact_uri(_EVAL_TABLE_FILE_NAME)
-        )
-
-    def _calculate_rouge(self):
-        try:
-            import evaluate
-
-            rouge = evaluate.load("rouge")
-        except Exception as e:
-            _logger.warning(
-                f"Failed to load 'rouge' metric (error: {e!r}), skipping metric logging."
-            )
-            return
 
-        predictions = (
-            self.y_pred.squeeze() if isinstance(self.y_pred, pd.DataFrame) else self.y_pred
-        )
-        metrics = rouge.compute(predictions=predictions, references=self.y, use_aggregator=False)
-        self.metrics_dict.update(metrics)
-        aggregate_metrics = rouge.compute(
-            predictions=predictions, references=self.y, use_aggregator=True
-        )
-        self.metrics.update(aggregate_metrics)
-
-    def _evaluate_text_summarization(self):
-        if self.dataset.has_targets:
-            self._calculate_rouge()
-        self._calculate_general_text_metrics()
-
-        self._log_eval_table()
-        name = _EVAL_TABLE_FILE_NAME.split(".", 1)[0]
+        columns = {}
+        for metric_name, metric_value in self.metrics_values.items():
+            scores = metric_value.scores
+            justifications = metric_value.justifications
+
+            if scores:
+                columns[f"{metric_name}/score"] = scores
+            if justifications:
+                columns[f"{metric_name}/justification"] = justifications
+        data = data.assign(**columns)
+        if metric_prefix:
+            artifact_file_name = f"{metric_prefix}{_EVAL_TABLE_FILE_NAME}"
+        else:
+            artifact_file_name = _EVAL_TABLE_FILE_NAME
+        mlflow.log_table(data, artifact_file=artifact_file_name)
+        name = artifact_file_name.split(".", 1)[0]
         self.artifacts[name] = JsonEvaluationArtifact(
-            uri=mlflow.get_artifact_uri(_EVAL_TABLE_FILE_NAME)
+            uri=mlflow.get_artifact_uri(artifact_file_name)
         )
 
-    def _evaluate_text(self):
-        self._calculate_general_text_metrics()
-        self._log_eval_table()
-        name = _EVAL_TABLE_FILE_NAME.split(".", 1)[0]
-        self.artifacts[name] = JsonEvaluationArtifact(
-            uri=mlflow.get_artifact_uri(_EVAL_TABLE_FILE_NAME)
-        )
+    def _update_metrics(self):
+        self.metrics = {}
+        for metric_name, metric_value in self.metrics_values.items():
+            if metric_value.aggregate_results:
+                for agg_name, agg_value in metric_value.aggregate_results.items():
+                    if agg_name == metric_name.split("/")[0]:
+                        self.metrics[metric_name] = agg_value
+                    else:
+                        self.metrics[f"{metric_name}/{agg_name}"] = agg_value
```

Comment:
Does the metric name look like `accuracy/sum`?

Reply:
With version it will be {metric_name}/{version}/{aggregate_name}, without version yes

Reply:
For aggregate metrics with identical names as the individual metrics, we assign the aggregate name key to match the metric name. The split is to remove the version from metric_name when we do the check.

Reply:
Got it. What would happen if I have a custom metric whose name is `my/best/metric`?

Reply:
Will discuss this and fix in follow-up if necessary!